src/com/lftechnology/java/training/niraj/ems/services/implementations/UserServiceImpl.java

```diff
@@ -0,0 +1,507 @@
+            }
+        }
+        if (userInfo.get(Constants.STATUS) == null) {
+            userInfo.put(Constants.STATUS, Status.ACTIVE.getStatus());
+        }
+        if (userInfo.get(Constants.IS_TERMINATED) == null) {
+            userInfo.put(Constants.IS_TERMINATED, Status.INACTIVE.getStatus());
+        }
+    }
+
+    /**
+     * Sets attributes
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * 
+     * @param employee
+     * @param userInfo
+     * @return
+     */
+    public Employee setAttributes(Employee employee, Map<String, String> userInfo) {
+        employeeDao.setEmployeeObject(userInfo, employee);
+        return employee;
+
+    }
+
+    /**
+     * Searches user
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param scanner
+     * @param console
+     * @return
+     * @throws SQLException
+     */
+    public String searchUser(Scanner scanner, Console console) throws SQLException {
+        Map<String, String> searchCondition = new LinkedHashMap<String, String>();
+        String username = userInput.getInput(scanner, console, Constants.SEARCH_BY_USERNAME);
+        searchCondition.put(Constants.USERNAME, username);
+        searchCondition.put(Constants.FULLNAME, username);
+        searchCondition.put(Constants.DEPARTMENT, username);
+        searchCondition.put(Constants.ADDRESS, username);
+        searchCondition.put(Constants.OPERATOR, Operators.AND.getOperator());
+        searchCondition.put(Constants.STATUS, Status.ACTIVE.getStatus());
+        List<Employee> searchResults = employeeDao.findAll(searchCondition, Operators.OR);
+        if (searchResults.size() == 0) {
+            return Constants.USER_NOT_FOUND;
+        }
+        return getFormattedSearchResults(searchResults);
+    }
+
+    /**
+     * Edits user profile
+     * 
+     * @author Niraj Rajbhandari<nirajrajbhandari@lftechnology.com>
+     * @param employee
+     * @param scanner
+     * @param console
+     * @throws SQLException
+     */
+    public void editUserProfile(Employee employee, Scanner scanner, Console console) throws SQLException {
+        List<String> choices = userInput.getMultipleChoices(scanner, console, Constants.EDIT_OPTION_LABEL, 1, 4);
+        Map<String, String> editInfo = getEditInfo(scanner, console, choices);
+        Employee editedEmployee = setAttributes(employee, editInfo);
+        save(editedEmployee);
+    }
+
+    /**
+     * Gets users list
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @return
+     * @throws SQLException
+     */
+    public String getUsersList() throws SQLException {
+        Map<String, String> searchCondition = new LinkedHashMap<String, String>();
+        searchCondition.put(Constants.STATUS, Status.ACTIVE.getStatus());
+        List<Employee> searchResults = employeeDao.findAll(searchCondition);
+        if (searchResults.size() == 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1155)